test/testfunctions.cpp

```diff
@@ -1373,6 +1373,9 @@ class TestFunctions : public TestFixture {
         check("int f() {}");
         ASSERT_EQUALS("[test.cpp:1]: (error) Found a exit path from function with non-void return type that has missing return statement\n", errout.str());
 
+        check("int main(void) {}");
```

Comment:
so for a c++ file there will be no warning? I would suggest that you test both with c++ and c:

    {
        const char code[] = "int main(void) {}";
        Settings s;

        s.standards.c = Standards::C89;
        check(code, "test.c", &s);  // c code
        ASSERT_EQUALS("..", errout.str());

        s.standards.c = Standards::C99;
        check(code, "test.c", &s);  // c code
        ASSERT_EQUALS("..", errout.str());

        check(code, "test.cpp", &s);  // c++ code
        ASSERT_EQUALS("..", errout.str());
    }

Reply:
That is the intention to have no warning for C++. I read that in C++ there is this behavior since the first standard from 1998.

`If control reaches the end of main without encountering a return statement, the effect is that of executing return 0;`

But I personally could not directly access the C++03 standard, so I have this information from secondary sources only.

Reply:
ok

Reply:
The `return` can be omitted for C++98 or later, or C99 or later. We should only warn when using C89. See cppreference for [C++ Main function](https://en.cppreference.com/w/cpp/language/main_function#Explanation), and [C Main function](https://en.cppreference.com/w/c/language/main_function#Explanation).

Reply:
great, thanks for the confirmation